diff --git a/src/pkcs8.rs b/src/pkcs8.rs
index 7b11c65..a3b9a0d 100644
--- a/src/pkcs8.rs
+++ b/src/pkcs8.rs
@@ -15,7 +15,7 @@
 //! breaking changes when using this module.
 
 pub use pkcs8::{
-    DecodePrivateKey, DecodePublicKey, Error, ObjectIdentifier, PrivateKeyInfoRef, Result, spki,
+    DecodePrivateKey, DecodePublicKey, Error, KeyError, ObjectIdentifier, PrivateKeyInfoRef, Result, spki,
 };
 
 #[cfg(feature = "alloc")]
@@ -162,14 +162,14 @@ impl TryFrom<PrivateKeyInfoRef<'_>> for KeypairBytes {
         // - 0x04: OCTET STRING tag
         // - 0x39: 57-byte length
         let secret_key = match private_key.private_key.as_bytes() {
-            [0x04, 0x39, rest @ ..] => rest.try_into().map_err(|_| Error::KeyMalformed),
-            _ => Err(Error::KeyMalformed),
+            [0x04, 0x39, rest @ ..] => rest.try_into().map_err(|_| KeyError::Invalid),
+            _ => Err(KeyError::Invalid),
         }?;
 
         let public_key = private_key
             .public_key
             .and_then(|bs| bs.as_bytes())
-            .map(|bytes| bytes.try_into().map_err(|_| Error::KeyMalformed))
+            .map(|bytes| bytes.try_into().map_err(|_| KeyError::Invalid))
             .transpose()?
             .map(PublicKeyBytes);
 
