diff --git a/src/sign/signing_key.rs b/src/sign/signing_key.rs
index 9f52147..79c0572 100644
--- a/src/sign/signing_key.rs
+++ b/src/sign/signing_key.rs
@@ -366,20 +366,20 @@ impl TryFrom<pkcs8::PrivateKeyInfoRef<'_>> for KeypairBytes {
     fn try_from(value: pkcs8::PrivateKeyInfoRef<'_>) -> Result<Self, Self::Error> {
         if value.algorithm.oid != super::ALGORITHM_OID {
-            return Err(pkcs8::Error::KeyMalformed);
+            return Err(pkcs8::KeyError::Invalid.into());
         }
         if value.private_key.as_bytes().len() != SECRET_KEY_LENGTH {
-            return Err(pkcs8::Error::KeyMalformed);
+            return Err(pkcs8::KeyError::Invalid.into());
         }
         let mut secret_key = [0u8; SECRET_KEY_LENGTH];
         secret_key.copy_from_slice(value.private_key.as_bytes());
         let verifying_key = if let Some(public_key) = value.public_key {
             if public_key.has_unused_bits() {
-                return Err(pkcs8::Error::KeyMalformed);
+                return Err(pkcs8::KeyError::Invalid.into());
             }
             let public_key = public_key.raw_bytes();
             if public_key.len() != PUBLIC_KEY_LENGTH {
-                return Err(pkcs8::Error::KeyMalformed);
+                return Err(pkcs8::KeyError::Invalid.into());
             }
             let mut bytes = [0u8; PUBLIC_KEY_LENGTH];
             bytes.copy_from_slice(public_key);
            Some(bytes)
@@ -409,11 +409,11 @@ impl TryFrom<&KeypairBytes> for SigningKey {
         let signing_key =
             SigningKey::from(SecretKey::try_from(&value.secret_key[..]).expect("invalid length"));
 
         if let Some(public_bytes) = value.verifying_key {
             let verifying_key =
-                VerifyingKey::from_bytes(&public_bytes).map_err(|_| pkcs8::Error::KeyMalformed)?;
+                VerifyingKey::from_bytes(&public_bytes).map_err(|_| pkcs8::KeyError::Invalid)?;
             if signing_key.verifying_key() != verifying_key {
-                return Err(pkcs8::Error::KeyMalformed);
+                return Err(pkcs8::KeyError::Invalid.into());
             }
         }
         Ok(signing_key)
