diff --git a/Cargo.toml b/Cargo.toml
index 64200b1..3d6a135 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -72,7 +72,7 @@ version = "1.0.117"
 version = "1.0.40"

 [build-dependencies.home]
-version = "=0.5.5"
+version = "0.5.12"
 default-features = false

 [build-dependencies.prost-build]
diff --git a/Cargo.toml.orig b/Cargo.toml.orig
index 66c0d4a..fa66269 100644
--- a/Cargo.toml.orig
+++ b/Cargo.toml.orig
@@ -28,5 +28,5 @@ serde_json = { workspace = true }
 thiserror = { workspace = true }

 [build-dependencies]
-home = { version = "=0.5.5", default-features = false }
+home = { version = "0.5.12", default-features = false }
 prost-build = "0.13.5"
