diff --git a/Cargo.toml b/Cargo.toml
index 89011f3..08d8fe3 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -173,10 +173,10 @@ features = [
 ]
 
 [target.'cfg(target_os = "windows")'.dependencies.webview2-com]
-version = "0.38"
+version = "0.39"
 
 [target.'cfg(target_os = "windows")'.dependencies.windows]
-version = "0.61"
+version = "0.62"
 features = [
     "Win32_System_Com",
     "Win32_System_Com_StructuredStorage",
@@ -184,7 +184,7 @@ features = [
 ]
 
 [target.'cfg(target_os = "windows")'.dependencies.windows-core]
-version = "0.61"
+version = "0.62"
 
 [lints.clippy]
 too_many_lines = "warn"
diff --git a/Cargo.toml.orig b/Cargo.toml.orig
index d84d3a4..9f1857c 100644
--- a/Cargo.toml.orig
+++ b/Cargo.toml.orig
@@ -61,9 +61,9 @@ objc2-ui-kit = { version = "0.2", features = ["UIImage"] }
 objc2-web-kit = { version = "0.2", features = ["WKSnapshotConfiguration", "block2"] }
 
 [target.'cfg(target_os = "windows")'.dependencies]
-webview2-com = "0.38"
-windows = { version = "0.61", features = ["Win32_System_Com", "Win32_System_Com_StructuredStorage", "Win32_Foundation"] }
-windows-core = "0.61"
+webview2-com = "0.39"
+windows = { version = "0.62", features = ["Win32_System_Com", "Win32_System_Com_StructuredStorage", "Win32_Foundation"] }
+windows-core = "0.62"
 
 [target.'cfg(target_os = "linux")'.dependencies]
 webkit2gtk = "2.0"
