diff --git a/build.rs b/build.rs
index 6cc1f0d1..1e7d15c7 100644
--- a/build.rs
+++ b/build.rs
@@ -860,27 +860,4 @@ fn check_features(
         }
     }
 
-    for &(lib, begin_version_major, end_version_major, begin_version_minor, end_version_minor) in
-        version_check_info.iter()
-    {
-        for version_major in begin_version_major..end_version_major {
-            for version_minor in begin_version_minor..end_version_minor {
-                let search_str =
-                    format!("[{lib}_version_greater_than_{version_major}_{version_minor}]");
-                let pos = stdout
-                    .find(&search_str)
-                    .expect("Variable not found in output")
-                    + search_str.len();
-
-                if &stdout[pos..pos + 1] == "1" {
-                    println!(
-                        r#"cargo:rustc-cfg=feature="{}""#,
-                        &search_str[1..(search_str.len() - 1)]
-                    );
-                    println!(r#"cargo:{}=true"#, &search_str[1..(search_str.len() - 1)]);
-                }
-            }
-        }
-    }
-
     let ffmpeg_lavc_versions = [
         ("ffmpeg_3_0", 57, 24),
         ("ffmpeg_3_1", 57, 48),
