diff --git a/src/private_key_info.rs b/src/private_key_info.rs
index e41a7a28..9ee713ac 100644
--- a/src/private_key_info.rs
+++ b/src/private_key_info.rs
@@ -203,7 +203,7 @@ where
         let private_key = Key::decode(reader)?;
 
         let _attributes =
-            reader.context_specific::<SequenceRef<'_>>(ATTRIBUTES_TAG, TagMode::Implicit)?;
+            reader.context_specific::<&SequenceRef>(ATTRIBUTES_TAG, TagMode::Implicit)?;
 
         let public_key = reader.context_specific::<PubKey>(PUBLIC_KEY_TAG, TagMode::Implicit)?;
 
