diff --git a/src/block_api.rs b/src/block_api.rs
index 4f612905..3cd39ae1 100644
--- a/src/block_api.rs
+++ b/src/block_api.rs
@@ -7,8 +7,8 @@ use digest::{
         AlgorithmName, Block, BlockSizeUser, Buffer, BufferKindUser, Eager, ExtendableOutputCore,
         FixedOutputCore, OutputSizeUser, Reset, UpdateCore, XofReaderCore,
     },
+    block_buffer::BlockSizes,
     common::{
-        BlockSizes,
         hazmat::{DeserializeStateError, SerializableState, SerializedState},
     },
     typenum::{IsLessOrEqual, True, U0, U200},
