diff --git a/build.rs b/build.rs
index 92e5c9b..a34b06c 100644
--- a/build.rs
+++ b/build.rs
@@ -898,8 +898,13 @@ fn print_prebuilt_src_binding_path() {
     download_file(&url, &src_binding_path);
   }
 
+  let out_binding_path = PathBuf::from(env::var_os("OUT_DIR").unwrap())
+    .join(name);
+  fs::copy(&src_binding_path, &out_binding_path)
+    .expect("failed to copy prebuilt src binding into OUT_DIR");
+
   println!(
     "cargo:rustc-env=RUSTY_V8_SRC_BINDING_PATH={}",
-    src_binding_path.display()
+    out_binding_path.display()
   );
 }
 
